website/www/site/content/en/documentation/programming-guide.md

```diff
@@ -5431,4 +5431,282 @@ use case.
+
+To create an SDK wrapper for use in a Python pipeline, do the following:
+
+1. Create a Python module for your cross-language transform.
+2. In the module, build the payload that should be used to initiate the cross-language transform expansion request using one of the available [PayloadBuilder](https://github.com/apache/beam/blob/master/sdks/python/apache_beam/transforms/external.py) classes.
+
+    The parameter names and types of the payload should map to parameter names and types of the configuration POJO provided to the Java `ExternalTransformBuilder`. Parameter types are mapped across SDKs using a [Beam schema](https://github.com/apache/beam/blob/master/model/pipeline/src/main/proto/schema.proto). Parameter names are mapped by simply converting Python underscore-separated variable names to camel-case (Java standard).
+
+    In the following example, kafka.py uses `NamedTupleBasedPayloadBuilder` to build the payload. The parameters map to the Java [KafkaIO.External.Configuration](https://github.com/apache/beam/blob/master/sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java) config object.
+
+    {{< highlight >}}
+ReadFromKafkaSchema = typing.NamedTuple(
+  'ReadFromKafkaSchema',
+  [
+      ('consumer_config', typing.Mapping[unicode, unicode]),
+      ('topics', typing.List[unicode])
+      # Other properties omitted for clarity.
+  ])
+payload = NamedTupleBasedPayloadBuilder(ReadFromKafkaSchema(...))
+    {{< /highlight >}}
+3. Start an expansion service unless one is specified by the pipeline creator. To use the `BeamJarExpansionService` utility from the Beam SDK, do the following:
+
+    1. Add a Gradle target to Beam that can be used to build a shaded expansion service JAR for the target Java transform. This target should produce a Beam JAR that contains all dependencies needed for expanding the Java transform and the JAR should be released with Beam.
+    2. In your Python module, instantiate `BeamJarExpansionService` with the Gradle target.
+
+        {{< highlight >}}
+    expansion_service = BeamJarExpansionService('sdks:java:io:expansion-service:shadowJar')
+        {{< /highlight >}}
+4. Add a Python wrapper transform class that extends [ExternalTransform](https://github.com/apache/beam/blob/master/sdks/python/apache_beam/transforms/external.py). Pass the payload and expansion service defined above as parameters to the constructor of the `ExternalTransform` parent class.
+
+#### 13.1.2. Creating cross-language Python transforms
+
+To make your Python transform usable with different SDK languages, you must create a wrapper class that registers an existing Python transform as a cross-language transform for use with the Python expansion service and calls into that existing transform to perform its intended operation.
+
+**Defining the wrapper class**
```

Comment:
I think we should call this something other than the "wrapper class" since we use that term for language wrappers defined to use cross-language transforms in other SDKs. How about "Python module" ?

Reply:
Yup, sounds good.